### DISCUSSION TITLE
Prevent autoscrolling when calling ui.log.push()

### DISCUSSION THREAD

**Discussion by jt48** [zauberzeug/nicegui#1281] [Ideas / Feature Requests]
### Question

Hi - Is there any way to prevent the ui.log element from scrolling right to the bottom when ui.log.push() is called?

I currently have my program logging frequently to the ui.log element and would like to able to scroll up to read the logs as it's being updated. Any help is appreciated.

Cheers - Josh

**Comment by falkoschindler** [comment]
Hi @jt48!
Currently it's not possible to prevent autoscrolling. But it would certainly be a useful feature. I'll convert this discussion into a feature request.

**Comment by CrimsonGhost69** [comment]
I also have my program logging to an ui.log() element. I used the code of the ui.log documentation and modified it with a formatter:

```python
class LogElementHandler(logging.Handler):
    """A logging handler that emits messages to a log element."""

    def __init__(self, element: ui.log, level: int = logging.NOTSET) -> None:
        self.element = element
        super().__init__(level)
        self.formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s', datefmt='%Y-%m-%dT%H:%M:%S%z')

    def emit(self, record: logging.LogRecord) -> None:
        try:
            msg = self.format(record)
            self.element.push(msg)
        except Exception:
            self.handleError(record)
```

My problem is the other way around. This means that autoscrolling no longer works for me with this example.
Any suggestions ?

**Reply by falkoschindler** [reply]
@CrimsonGhost69 Can you, please, create a new Q&A including a minimal executable example? That would allow us to help more efficiently. Thanks!
